RecoPixelVertexing/PixelVertexFinding/plugins/PixelVertexCollectionTrimmer.cc

```diff
@@ -1,128 +1,100 @@
 public:
   explicit PixelVertexCollectionTrimmer(const edm::ParameterSet&);
-  ~PixelVertexCollectionTrimmer() override;
 
   static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
 
 private:
   void produce(edm::Event&, const edm::EventSetup&) override;
 
-  edm::EDGetTokenT<reco::VertexCollection> vtxToken_;
-  unsigned int maxVtx_;
-  double fractionSumPt2_;
-  double minSumPt2_;
+  edm::EDGetTokenT<reco::VertexCollection> const vtxToken_;
+  uint const maxVtx_;
+  double const fractionSumPt2_;
+  double const minSumPt2_;
 
-  PVClusterComparer* pvComparer_;
+  std::unique_ptr<PVClusterComparer> pvComparer_;
 };
 
-PixelVertexCollectionTrimmer::PixelVertexCollectionTrimmer(const edm::ParameterSet& iConfig) {
-  edm::InputTag vtxInputTag = iConfig.getParameter<edm::InputTag>("src");
-  vtxToken_ = consumes<reco::VertexCollection>(vtxInputTag);
-  maxVtx_ = iConfig.getParameter<unsigned int>("maxVtx");
-  fractionSumPt2_ = iConfig.getParameter<double>("fractionSumPt2");
-  minSumPt2_ = iConfig.getParameter<double>("minSumPt2");
-
-  edm::ParameterSet PVcomparerPSet = iConfig.getParameter<edm::ParameterSet>("PVcomparer");
-  double track_pt_min = PVcomparerPSet.getParameter<double>("track_pt_min");
-  double track_pt_max = PVcomparerPSet.getParameter<double>("track_pt_max");
-  double track_chi2_max = PVcomparerPSet.getParameter<double>("track_chi2_max");
-  double track_prob_min = PVcomparerPSet.getParameter<double>("track_prob_min");
-
-  pvComparer_ = new PVClusterComparer(track_pt_min, track_pt_max, track_chi2_max, track_prob_min);
+PixelVertexCollectionTrimmer::PixelVertexCollectionTrimmer(const edm::ParameterSet& iConfig)
+    : vtxToken_(consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("src"))),
+      maxVtx_(iConfig.getParameter<uint>("maxVtx")),
+      fractionSumPt2_(iConfig.getParameter<double>("fractionSumPt2")),
+      minSumPt2_(iConfig.getParameter<double>("minSumPt2")) {
+  if (fractionSumPt2_ < 0)
+    edm::LogWarning("Configuration") << "value of \"fractionSumPt2\" is negative.";
+  else if (fractionSumPt2_ > 1)
+    edm::LogWarning("Configuration") << "value of \"fractionSumPt2\" is larger than 1.";
+
+  auto const PVcomparerPSet = iConfig.getParameter<edm::ParameterSet>("PVcomparer");
+  auto const track_pt_min = PVcomparerPSet.getParameter<double>("track_pt_min");
+  auto const track_pt_max = PVcomparerPSet.getParameter<double>("track_pt_max");
+  auto const track_chi2_max = PVcomparerPSet.getParameter<double>("track_chi2_max");
+  auto const track_prob_min = PVcomparerPSet.getParameter<double>("track_prob_min");
+
+  if (track_pt_min >= track_pt_max)
+    edm::LogWarning("Configuration") << "PVcomparer.track_pt_min (" << track_pt_min << ") >= PVcomparer.track_pt_max ("
+                                     << track_pt_max << ") : PVClusterComparer will use pT=" << track_pt_max
+                                     << " for all selected tracks.";
```

Comment:
wouldn't it make sense to throw here as well?

Reply:
Done.